### ISSUE TITLE
Why do different graphics cards use different amounts of VRAM for the same model?

### ISSUE THREAD

**Issue by xianfu** [ultralytics/ultralytics#21416]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

1080TI

<img width="826" height="89" alt="Image" src="https://github.com/user-attachments/assets/2e8e9a89-ca1e-4409-bf7a-cfd34e0318cf" />

P100

<img width="644" height="271" alt="Image" src="https://github.com/user-attachments/assets/ddbe0321-de3d-4f40-82b6-2d92c639d3ff" />

**Comment by UltralyticsAssistant** [comment]
👋 Hello @xianfu, thank you for reaching out and sharing your question with the Ultralytics team 🚀! For new users, we suggest visiting the [Docs](https://docs.ultralytics.com/) where you'll find helpful [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) examples, as well as answers to many common questions.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us investigate further.

If you have a ❓ Question about custom training or environment differences, please include specific details such as your environment setup, command outputs, and any logs to help us assist you efficiently. Also, have a look at our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/) to optimize your workflow.

You’re welcome to join the Ultralytics community where you prefer! For real-time conversation, hop onto [Discord](https://discord.com/invite/ultralytics) 🎧. For in-depth topics, try [Discourse](https://community.ultralytics.com/). Or participate in discussions over on our [Subreddit](https://www.reddit.com/r/Ultralytics/).

## Upgrade

Please ensure you are using the latest `ultralytics` package and all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to confirm the issue persists in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO can be run in any of these up-to-date, verified environments (with dependencies, [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/), [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are passing. CI tests verify correct operation for all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

---

This is an automated response to help guide you efficiently. An Ultralytics engineer will review your question and provide further assistance soon 😊.

**Comment by glenn-jocher** [comment]
Different GPUs use varying amounts of VRAM for the same model due to several hardware and software factors. The P100 and 1080TI have different memory architectures, CUDA compute capabilities, and driver optimizations that affect memory allocation strategies. Additionally, memory alignment requirements, fragmentation patterns, and hardware-specific optimizations vary between GPU generations and manufacturers, leading to different memory footprints even when running identical models.

You can monitor GPU memory usage programmatically using `torch.cuda.memory_reserved()` or the [cuda_memory_usage function](https://docs.ultralytics.com/reference/utils/torch_utils/#ultralytics.utils.torch_utils.cuda_memory_usage) in the Ultralytics codebase for detailed memory profiling across different devices.
